megamek/src/megamek/client/ui/swing/RandomArmyDialog.java

```diff
@@ -536,20 +537,25 @@ public RandomArmyDialog(ClientGUI cl) {
         m_pRightPane.add(m_pPreview, CARD_PREVIEW);
         m_pRightPane.add(m_pForceGen.getRightPanel(), CARD_FORCE_TREE);
         
-        m_pSplit = new javax.swing.JSplitPane(javax.swing.JSplitPane.HORIZONTAL_SPLIT,m_pMain, m_pRightPane);
+        m_pSplit = new JSplitPane(JSplitPane.HORIZONTAL_SPLIT, m_pMain, m_pRightPane);
         m_pSplit.setOneTouchExpandable(false);
         m_pSplit.setResizeWeight(0.5);
 
         // construct the main dialog
         setLayout(new BorderLayout());
-        this.setPreferredSize(new Dimension(800,500));
+//        this.setPreferredSize(new Dimension(800,500));
```

Comment:
Why's this commented out, along with the pack() on line 550?

Reply:
The whole dialog now remembers its position and size, therefore the setpreferredsize and pack are unnecessary. I'll remove the lines.